parachain/pallets/assets/Cargo.toml

```diff
@@ -8,22 +8,23 @@ edition = "2018"
 targets = ["x86_64-unknown-linux-gnu"]
 
 [dependencies]
-codec = { version = "2.0.0", package = "parity-scale-codec", default-features = false, features = ["derive"] }
+serde = { version = "1.0.130", optional = true }
+codec = { version = "2.2.0", package = "parity-scale-codec", default-features = false, features = ["derive"] }
+scale-info = { version = "1.0", default-features = false, features = ["derive"] }
```

Comment:
Wherever you add `scale-info` as a dependency, you also need to include its `std` [feature](https://github.com/paritytech/scale-info/blob/d13be2a6437493a0ef8ed5a471c16abcc3e424a5/Cargo.toml#L26) in the `std` feature of  the crates where it is used (See further below in this file). This is all to ensure that conditional compilation continues to work. See https://doc.rust-lang.org/cargo/reference/features.html for more details.

The `std` feature is used by Parity to compile code to work with Rust's standard library. However parts of a Substrate blockchain, like its runtime code are also [compiled to WASM](https://docs.substrate.io/v3/advanced/executor/#wasm-execution) where the standard library is not available.